sphinx_gallery/tests/test_recommender.py

```diff
@@ -0,0 +1,143 @@
+# Author: Arturo Amor
+# License: 3-clause BSD
+"""Test the example recommender plugin."""
+
+import codecs
+import os
+import os.path as op
+import re
+import shutil
+from io import StringIO
+
+import pytest
+
+import sphinx_gallery.gen_rst as sg
+from sphinx.application import Sphinx
+from sphinx.util.docutils import docutils_namespace
+from sphinx_gallery.recommender import ExampleRecommender, _write_recommendations
+
+
+@pytest.fixture(scope="module")
+def sphinx_app(tmpdir_factory, req_mpl, req_pil):
+    return _sphinx_app(tmpdir_factory, "html")
+
+
+def _sphinx_app(tmpdir_factory, buildername):
+    # Skip if numpy not installed
+    pytest.importorskip("numpy")
+
+    temp_dir = (tmpdir_factory.getbasetemp() / f"root_{buildername}").strpath
+    src_dir = op.join(op.dirname(__file__), "tinybuild")
+
+    def ignore(src, names):
+        return ("_build", "gen_modules", "auto_examples")
+
+    shutil.copytree(src_dir, temp_dir, ignore=ignore, dirs_exist_ok=True)
+    # For testing iteration, you can get similar behavior just doing `make`
+    # inside the tinybuild/doc directory
+    conf_dir = op.join(temp_dir, "doc")
+    out_dir = op.join(conf_dir, "_build", buildername)
+    toctrees_dir = op.join(conf_dir, "_build", "toctrees")
+    # Avoid warnings about re-registration, see:
+    # https://github.com/sphinx-doc/sphinx/issues/5038
+    with docutils_namespace():
+        app = Sphinx(
+            conf_dir,
+            conf_dir,
+            out_dir,
+            toctrees_dir,
+            buildername=buildername,
+            status=StringIO(),
+            warning=StringIO(),
+        )
+        # need to build within the context manager
+        # for automodule and backrefs to work
+        app.build(False, [])
+    return app
+
+
+def test_recommend_n_examples(sphinx_app):
```

Comment:
So sorry, I should have been clearer, this test can stay in `test_full.py`. Since it uses `sphinx_app` it may as well stay in `test_full.py`, I was meaning more the other 2 tests, which don't need `spinx_app`.

Also thank you for separating `test_example_recommender_methods` to it's own unit test.